presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -1156,6 +1160,18 @@ protected Object visitSubscriptExpression(SubscriptExpression node, Object conte
                 return new SubscriptExpression(toExpression(base, type(node.getBase())), toExpression(index, type(node.getIndex())));
             }
 
+            // Subscript on Row hasn't got a dedicated operator. It is interpreted by hand.
+            if (base instanceof SingleRowBlock) {
```

Comment:
This is unexpected... A dup code logic is added to the interpreter. I wasn't aware of the change. Maybe we should add the same logic to `RowExpressionInterpreter` and enable the test. cc: @wenleix

Reply:
We had a hand written logic just to handle subscript previously. We definitely should handle that in `RowExpressionInterpreter` as well.

Ideally, we need to force desugar for all expressions, if you want to dig deeper. Adding to planOptimizer will not help tests etc. where planOptimizers are only explicitly called.